include/pybind11/pytypes.h

```diff
@@ -426,6 +490,14 @@ class PYBIND11_EXPORT_EXCEPTION error_already_set : public std::runtime_error {
 
 private:
     object m_type, m_value, m_trace;
+    mutable std::string m_lazy_what;
+
+    static const char *get_class_name(PyObject *py) {
```

Comment:
I'd be find making this a detail level function too, but I don't think it should go in as a public func especially with such a generic name.

Reply:
It's a very small but time-tested and very useful function that I copied from here:

https://github.com/google/clif/blob/8c574ffc600f5ef44da3624c9315704a778a039b/clif/python/runtime.cc#L159

I agree a better name would be good. `private` is essential for guarding invariants, using `private` to hide things just because is less useful.

Maybe (in namespace detail):
```
// obj.__class__.__name__
inline const char *obj_class_name(PyObject *obj)
```
?
```